File: backend/internal/models/truck.go
package models

import "time"

const (
	TruckStatusIdle        = "idle"
	TruckStatusEnRoute     = "en_route"
	TruckStatusFull        = "full"
	TruckStatusMaintenance = "maintenance"
)

type Truck struct {
	ID                 string    `json:"id"`
	RegistrationNumber string    `json:"registration_number"`
	CapacityKg         float64   `json:"capacity_kg"`
	DriverID           *string   `json:"driver_id"`
	CurrentLat         float64   `json:"current_lat"`
	CurrentLng         float64   `json:"current_lng"`
	Status             string    `json:"status"`
	CreatedAt          time.Time `json:"created_at"`
}

type TruckInput struct {
	RegistrationNumber string  `json:"registration_number"`
	CapacityKg         float64 `json:"capacity_kg"`
	DriverID           *string `json:"driver_id"`
	CurrentLat         float64 `json:"current_lat"`
	CurrentLng         float64 `json:"current_lng"`
	Status             string  `json:"status"`
}

File: frontend/src/types/truck.js
/**
 * @typedef {Object} Truck
 * @property {string} id
 * @property {string} registration_number
 * @property {number} capacity_kg
 * @property {string|null} driver_id
 * @property {number} current_lat
 * @property {number} current_lng
 * @property {'idle'|'en_route'|'full'|'maintenance'} status
 * @property {string} created_at
 * @property {User} [driver]
 */

/**
 * @typedef {Object} TruckInput
 * @property {string} registration_number
 * @property {number} capacity_kg
 * @property {string|null} [driver_id]
 * @property {number} [current_lat]
 * @property {number} [current_lng]
 * @property {string} [status]
 */

export const _types = null
